src/components/common/ListPeopleAndRoles.vue

```diff
@@ -146,19 +154,31 @@ import { PartyTypes, RoleTypes, RouteNames } from '@/enums'
 export default class ListPeopleAndRoles extends Vue {
   @Prop({ default: false }) readonly isSummary!: boolean
   @Prop({ default: false }) readonly disabled!: boolean
+  @Prop({ default: true }) readonly showDeliveryAddressColumn!: boolean
+  @Prop({ default: true }) readonly showRolesColumn!: boolean
+  @Prop({ default: false }) readonly showEmailColumn!: boolean
 
   @Getter isIncorporationFiling!: boolean
   @Getter isRegistrationFiling!: boolean
+  @Getter isFullRestorationFiling!: boolean
+  @Getter isLimitedRestorationFiling!: boolean
   @Getter getAddPeopleAndRoleStep!: PeopleAndRoleIF
   @Getter getShowErrors!: boolean
 
   // Enum for template
   readonly RouteNames = RouteNames
 
   // Local properties
-  readonly tableHeaders: Array<string> = ['Name', 'Mailing Address', 'Delivery Address', 'Roles']
   protected activeIndex: number // is NaN for new org/person
 
+  get tableHeaders (): Array<string> {
+    const headers = ['Name', 'Mailing Address']
+    if (this.showDeliveryAddressColumn) headers.push('Delivery Address')
+    if (this.showRolesColumn) headers.push('Roles')
+    if (this.showEmailColumn) headers.push('Email')
+    return headers
+  }
+
```

Comment:
What do you think about making a sub-component to display a row in the table?  Then, if I'm thinking about this correctly, these properties could be computed row attributes.

Reply:
I think you're right, but I think this isn't complicated enough, or used elsewhere, that it needs to be its own component IMHO. It's a gray area.